test_conformance/extensions/cl_khr_external_semaphore/test_external_semaphore.cpp

```diff
@@ -174,104 +203,124 @@ int test_external_semaphores_multi_context(cl_device_id deviceID,
+    test_error(err, "Failed to query export handle types");
 
-        clExternalExportableSemaphore sema_ext_1(
-            vkVk2CLSemaphore, context, vkExternalSemaphoreHandleType, deviceID);
-        clExternalExportableSemaphore sema_ext_2(vkVk2CLSemaphore, context2,
-                                                 vkExternalSemaphoreHandleType,
-                                                 deviceID);
+    // Find handles that support both import and export
+    std::unordered_set<cl_external_semaphore_handle_type_khr>
+        import_export_handle_types;
 
-        clCommandQueueWrapper queue1 =
-            clCreateCommandQueue(context, deviceID, 0, &err);
-        test_error(err, "Could not create command queue");
+    std::copy(import_handle_types.begin(), import_handle_types.end(),
+              std::inserter(import_export_handle_types,
+                            import_export_handle_types.end()));
+    std::copy(export_handle_types.begin(), export_handle_types.end(),
+              std::inserter(import_export_handle_types,
+                            import_export_handle_types.end()));
 
-        clCommandQueueWrapper queue2 =
-            clCreateCommandQueue(context2, deviceID, 0, &err);
-        test_error(err, "Could not create command queue");
-
-        // Signal semaphore 1 and 2
-        clEventWrapper signal_event;
-        err = clEnqueueSignalSemaphoresKHR(queue1, 1,
-                                           &sema_ext_1.getCLSemaphore(),
-                                           nullptr, 0, nullptr, &signal_event);
-        test_error(err, "Could not signal semaphore");
+    cl_context context2 =
+        clCreateContext(NULL, 1, &deviceID, notify_callback, NULL, &err);
+    test_error(err, "Failed to create context2");
 
-        // Wait semaphore 1
-        clEventWrapper wait_1_event;
-        err =
-            clEnqueueWaitSemaphoresKHR(queue1, 1, &sema_ext_1.getCLSemaphore(),
-                                       nullptr, 0, nullptr, &wait_1_event);
-        test_error(err, "Could not wait semaphore");
+    clCommandQueueWrapper queue1 =
+        clCreateCommandQueue(context, deviceID, 0, &err);
+    test_error(err, "Could not create command queue");
 
-        err = clEnqueueSignalSemaphoresKHR(queue2, 1,
-                                           &sema_ext_2.getCLSemaphore(),
-                                           nullptr, 0, nullptr, &signal_event);
-        test_error(err, "Could not signal semaphore");
+    clCommandQueueWrapper queue2 =
+        clCreateCommandQueue(context2, deviceID, 0, &err);
+    test_error(err, "Could not create command queue");
 
-        // Wait semaphore 2
-        clEventWrapper wait_2_event;
-        err =
-            clEnqueueWaitSemaphoresKHR(queue2, 1, &sema_ext_2.getCLSemaphore(),
-                                       nullptr, 0, nullptr, &wait_2_event);
-        test_error(err, "Could not wait semaphore");
+    if (import_export_handle_types.empty())
+    {
```

Comment:
This doesn't seem to be a strict requirement. There are different configs in which we could test this scenario.
1. One OpenCL context exporting the semaphore and the other OpenCL context importing the semaphore. --> This would require both exportable and importable semaphores supported.
2. Use Vulkan as intermediate source for creating semaphores, two OpenCL contexts importing two semaphores from Vulkan and communicating across the two contexts using the imported semaphores.

SyncFD would require both exportable and importable semaphore support irrespective of above modes due to additional limitations to signal and wait.